flink-ml-lib/src/main/java/org/apache/flink/ml/feature/univariatefeatureselector/UnivariateFeatureSelectorModel.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+
+        private final String inputCol;
+        private final String broadcastKey;
+        private int[] indices;
+
+        public PredictOutputFunction(String inputCol, String broadcastKey) {
+            this.inputCol = inputCol;
+            this.broadcastKey = broadcastKey;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (indices == null) {
+                UnivariateFeatureSelectorModelData modelData =
+                        (UnivariateFeatureSelectorModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+                indices = Arrays.stream(modelData.indices).sorted().toArray();
+            }
+
+            if (indices.length == 0) {
+                return Row.join(row, Row.of(Vectors.dense()));
+            } else {
+                Vector inputVec = ((Vector) row.getField(inputCol));
+                Preconditions.checkArgument(
+                        inputVec.size() > indices[indices.length - 1],
+                        "Input %s features, but UnivariateFeatureSelector is "
+                                + "expecting at least %s features as input.",
+                        inputVec.size(),
+                        indices[indices.length - 1] + 1);
+                Vector outputVec = selectByIndices(inputVec, indices);
+                return Row.join(row, Row.of(outputVec));
+            }
+        }
+
+        /**
+         * Selects a subset of the vector base on the indices. Note that the input indices must be
+         * sorted in ascending order if the input vector is sparse.
+         */
+        private Vector selectByIndices(Vector vector, int[] selectedIndices) {
+            if (vector instanceof DenseVector) {
+                DenseVector resultVec = new DenseVector(selectedIndices.length);
+                for (int i = 0; i < selectedIndices.length; i++) {
+                    resultVec.set(i, vector.get(selectedIndices[i]));
+                }
+                return resultVec;
+            } else {
+                List<Integer> resultIndices = new ArrayList<>();
+                List<Double> resultValues = new ArrayList<>();
+
+                int[] indices = ((SparseVector) vector).indices;
+                for (int i = 0, j = 0; i < indices.length && j < selectedIndices.length; ) {
+                    if (indices[i] == selectedIndices[j]) {
+                        resultIndices.add(j++);
+                        resultValues.add(((SparseVector) vector).values[i++]);
+                    } else if (indices[i] > selectedIndices[j]) {
+                        j++;
+                    } else {
+                        i++;
+                    }
+                }
+                return new SparseVector(
```

Comment:
The SparseVector instance created here is guaranteed to be dense. Would it be simpler to just return DenseVector here?